### DISCUSSION TITLE
How to bring own custom background to gseapy

### DISCUSSION THREAD

**Discussion by Cher-HAN** [zqfang/GSEApy#177] [Q&A]
Hi zq,

Thanks so much for the useful package and congratulations to the gseapy paper.
I am doing an analysis of GO terms, and there are no significant enriched GO terms, and I guess it is because the default background(whole human protein-coding genes) includes many irrelevant GO terms. 
I could use a biological meaningful condition to filter the background annotation, and I plan to generate my own gmt annotation and then supply the gmt file to gseapy as custom background. 
However, I am curious whether I could input a list of gene names to gseapy, and it could automatically generate the custom background base on the gene list.

Kind regards,
Xueer

**Comment by zqfang** [comment]
What function you'er trying to use ?

if `gseapy.enrich()` or `gseapy.enrichr()`. Just set the `background` argument to your own input gene list 

see some explanation [here](https://gseapy.readthedocs.io/en/latest/gseapy_example.html#About-Background-genes)

**Reply by Cher-HAN** [reply]
I am using prerank.

**Reply by zqfang** [reply]
the background is your input ranking list for gsea, prerank, ssgsea

**Reply by Cher-HAN** [reply]
I think the background is "gene_sets". And is it possible that I input a list of genes to generate the custom a gene_sets?

**Reply by zqfang** [reply]
Of course. All you need is given `gene_sets` a dict or GMT file your own. 

check this [tutorial](https://gseapy.readthedocs.io/en/latest/gseapy_example.html#3.-Prerank-example) out
